mex-wholebodymodel/library/src/modelstate.cpp

```diff
@@ -20,62 +21,48 @@
 // global includes
 
 // library includes
-#include <mex.h>
-
 #include <iDynTree/ModelIO/URDFDofsImport.h>
-#include<yarpWholeBodyInterface/yarpWholeBodyModel.h>
+#include <yarpWholeBodyInterface/yarpWholeBodyModel.h>
 #include <yarpWholeBodyInterface/yarpWbiUtil.h>
-#include <yarp/os/Network.h>
 #include <yarp/os/ResourceFinder.h>
-#include <yarp/sig/Image.h>
 
 // local includes
 #include "modelstate.h"
 
 using namespace mexWBIComponent;
 
-ModelState* ModelState::modelState;
-wbi::iWholeBodyModel * ModelState::robotWBIModel = NULL;
+// ModelState *ModelState::modelState = NULL;
+boost::scoped_ptr<ModelState> ModelState::modelState(NULL);
```

Comment:
This is just a temporary solution, which makes the destructor "a bit" more stable, but when I have solved the problem of releasing the two allocated arrays. I can do it in another way.